conan/tools/env/environment.py

```diff
@@ -0,0 +1,269 @@
+        # FIXME: This is broken and doesnt work
+        deactivate = ""
+        capture = textwrap.dedent("""\
+            {deactivate}
+            """).format(deactivate=deactivate if generate_deactivate else "")
+        result = [capture]
+        for varname, varvalues in self._values.items():
+            value = self._format_value(varname, varvalues, "$env:{name}", pathsep)
+            result.append('$env:{}={}'.format(varname, value))
+
+        content = "\n".join(result)
+        save(filename, content)
+
+    def save_sh(self, filename, generate_deactivate=False, pathsep=os.pathsep):
+        deactivate = textwrap.dedent("""\
+            echo Capturing current environment in deactivate_{filename}
+            echo echo Restoring variables >> deactivate_{filename}
+            for v in {vars}
+            do
+                value=$(printenv $v)
+                if [ -n "$value" ]
+                then
+                    echo export "$v=$value" >> deactivate_{filename}
+                else
+                    echo unset $v >> deactivate_{filename}
+                fi
+            done
+            echo Configuring environment variables
+            """.format(filename=filename, vars=" ".join(self._values.keys())))
+        capture = textwrap.dedent("""\
+           {deactivate}
+           echo Configuring environment variables
+           """).format(deactivate=deactivate if generate_deactivate else "")
+        result = [capture]
+        for varname, varvalues in self._values.items():
+            value = self._format_value(varname, varvalues, "${name}", pathsep)
+            if value:
+                result.append('export {}="{}"'.format(varname, value))
+            else:
+                result.append('unset {}'.format(varname))
+
+        content = "\n".join(result)
+        save(filename, content)
+
+    def compose(self, other):
+        """
+        :type other: Environment
+        """
+        for k, v in other._values.items():
+            existing = self._values.get(k)
+            if existing is None:
+                self._values[k] = v
+            else:
+                try:
+                    index = v.index(_EnvVarPlaceHolder)
+                except ValueError:  # The other doesn't have placeholder, overwrites
+                    self._values[k] = v
+                else:
+                    new_value = v[:]  # do a copy
+                    new_value[index:index + 1] = existing  # replace the placeholder
+                    # Trim front and back separators
+                    val = new_value[0]
+                    if isinstance(val, _Sep) or val is _PathSep:
+                        new_value = new_value[1:]
+                    val = new_value[-1]
+                    if isinstance(val, _Sep) or val is _PathSep:
+                        new_value = new_value[:-1]
+                    self._values[k] = new_value
+        return self
+
+
+class ProfileEnvironment:
+    def __init__(self):
```

Comment:
may ctor receive arguments as loads? I see usage pattern is `x = ProfileEnvironment(); x.load(something)`

Reply:
Yes, lets change it to ``ProfileEnvirionment.loads(something)`` staticmethod, as it will be more aligned with the rest of the codebase.